src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.abstracts.Figure;
+import core.basesyntax.enums.Colors;
+import core.basesyntax.figures.*;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Circle circle = new Circle();
+    private final IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid();
+    private final Rectangle rectangle = new Rectangle();
+    private final RightTriangle rightTriangle = new RightTriangle();
+    private final Square square = new Square();
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int ALL_FIGURES = 5;
+    private static final int DEF_PARAMETER = 10;
+
+
+    public Figure getRandomFigure() {
+        Figure[] allFigures = {circle, isoscelesTrapezoid, rectangle, rightTriangle, square};
+        int trueRandom = random.nextInt(ALL_FIGURES);
+        switch (trueRandom) {
+            case 0 : {
+                circle.setRadius(random.nextInt(DEF_PARAMETER));
+                circle.setColor(colorSupplier.getRandomColor());
+                circle.draw();
+                return circle;
```

Comment:
1. Please create all-args constructors to simplify code
2. draw() method should be called in Main class after figure is added to array, not during generation
```suggestion
                return new Circle(random.nextInt(DEF_PARAMETER), colorSupplier.getRandomColor());
```